dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +109,183 @@ public Bundle getBundleById(String id) throws DotDataException {
+
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsFactory.deletePushedAssetsByBundle(bundleId);
+
+		Logger.info(this, "Removing all assets from bundle: " + bundleId);
+		this.bundleFactory.deleteAllAssetsFromBundle(bundleId);
+
+		Logger.info(this, "Removing environments from bundle: " + bundleId);
+		this.bundleFactory.deleteBundleEnvironmentByBundle(bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId);
+		this.deleteBundle(bundleId);
+	}
+
+	@WrapInTransaction
+	@Override
+	public void deleteBundleAndDependencies(final String bundleId, final User user) throws DotDataException {
+
+		final Bundle bundle = this.getBundleById(bundleId);
+		if (null != bundle) {
+
+			this.deleteBundleAndDependencies(bundle, user);
+		} else {
+
+			throw new NotFoundInDbException("The bundle id: " + bundleId + " does not exists");
+		}
+	}
+
+	private void validateBundleDeletePermission(final User user, final Bundle bundle) throws DotDataException {
+
+		/**
+		if (!APILocator.getPermissionAPI().doesUserHavePermission(bundle,
+				PermissionAPI.PERMISSION_EDIT, user, false)) {
+
+			throw new DotSecurityException("User : " + user.getUserId() +
+					", is not allowed to delete the bundle: " + bundleId)
+		}*/
+		// todo: bundle is not a permissionable yet, so can not validate yet.
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		int offset               = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		List<Bundle> sentBundles = isAdmin?
+				this.bundleFactory.findSentBundles(olderThan, limit, offset):
+				this.bundleFactory.findSentBundles(olderThan, user.getUserId(), limit, offset);
+
+		Logger.info(this, "Deleting bundles older than: " + olderThan);
+		while (UtilMethods.isSet(sentBundles)) {
+
+			for (final Bundle bundle : sentBundles) {
+
+				this.deleteBundleAndDependencies(bundle, user);
+				bundlesDeleted.add(bundle.getId());
+			}
+
+			sentBundles = isAdmin?
+					this.bundleFactory.findSentBundles(olderThan, limit, offset):
+					this.bundleFactory.findSentBundles(olderThan, user.getUserId(), limit, offset);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String>  deleteAllBundles(final User user,
+										 final PublishAuditStatus.Status ...statuses) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		final int offset         = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		final List<Status> statusList = Arrays.asList(statuses);
+		List<String> bundleIDs = isAdmin ? publishAuditAPI.getBundleIdByStatus(statusList,limit,offset) :
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)